### PR TITLE
Improve transaction submitter to avoid a sudden surge 

### PR DESCRIPTION
**Pull request by JeffreyDallas**
Changes

- Added a speeddometer to measure actual TPS of submitted transaction
- After a node offline a while, avoid it sends large number of transaction in short time by adding a upper limit

Test result

https://swirldslabs.slack.com/archives/C03G7CBJJ06/p1678822419139349

---

### Before

image

## After

image

### CODE REVIEW

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/TransactionSubmitter.java [hunk 1]
```diff
@@ -365,6 +365,22 @@ public boolean canSubmitMore(Platform platform) {
                 return true;
             }
         } else if (this.goal == SUBMIT_GOAL.TRANS_PER_SECOND_PER_NODE) {
+            // if the node is offline for a while due to network disruption,
+            // it may try to catch up the lost transactions count by submitting
+            // many transactions in short window and lead to a burst of transactions. This is not good for
+            // platform, so we need to check the current TPS and make sure it is not too high.
+            final double tarnSubTSP = (double) metrics.getValue("Debug.info", "tranSubTPS");
+            if (tarnSubTSP > tranPerSecondGoal * 1.3) {
+                //                logger.info(
```
Comment (hendrikebbers) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/TransactionSubmitter.java:374]: can you remove the commented code or maybe define it as debug logging?

**Review by hendrikebbers — APPROVED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/TransactionSubmitter.java (hunk 1 above)
Comment (JeffreyDallas) [reply platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/TransactionSubmitter.java]: Fixed

**Review by nathanklick — DISMISSED** [review]
Approved, the Snyk alerts may be ignored on this PR.

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/TransactionSubmitter.java [hunk 2]
```diff
@@ -365,6 +367,15 @@ public boolean canSubmitMore(Platform platform) {
                 return true;
             }
         } else if (this.goal == SUBMIT_GOAL.TRANS_PER_SECOND_PER_NODE) {
+            // if the node is offline for a while due to network disruption,
+            // it may try to catch up the lost transactions count by submitting
+            // many transactions in short window and lead to a burst of transactions. This is not good for
+            // platform, so we need to check the current TPS and make sure it is not too high.
+            final double tarnSubTSP = (double) metrics.getValue("Debug.info", "tranSubTPS");
+            if (tarnSubTSP > tranPerSecondGoal * ALLOWED_CATCHUP_DELTA) {
```
Comment (qnswirlds) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/TransactionSubmitter.java:375]:
```suggestion
            final double tranSubTPS = (double) metrics.getValue("Debug.info", "tranSubTPS");
            if (tranSubTPS > tranPerSecondGoal * ALLOWED_CATCHUP_DELTA) {
```

**Review by hendrikebbers — APPROVED** [review]

**Pull request merged**